### PR TITLE
#493 linkable versions in release notes, formatting changes

### PR DESCRIPTION
**Pull request by micd**
Changed formatting in release notes to have linkable versions.

### CODE REVIEW

File: subprojects/shipkit/src/test/groovy/org/shipkit/internal/notes/format/DetailedFormatterTest.groovy [hunk 1]
```diff
@@ -52,18 +54,19 @@ Release notes:
...
     def "formats header when emphasized version"() {
         expect:
-        DetailedFormatter.header("v0.1.0", new Date(1483500000000), false) == "**v0.1.0 (2017-01-04)** - "
+        DetailedFormatter.header("v0.1.0", false) == "#### v0.1.0"
     }
 
     def "formats header when regular version"() {
         expect:
-        DetailedFormatter.header("v0.1.0", new Date(1483500000000), true) == "# v0.1.0 (2017-01-04) - "
+        DetailedFormatter.header("v0.1.0", true) == "# v0.1.0"
```
Comment (wwilk) [inline subprojects/shipkit/src/test/groovy/org/shipkit/internal/notes/format/DetailedFormatterTest.groovy]: I guess now emphasized version would be "#", and a regular one "####". What do you think @mockitoguy ?

File: subprojects/shipkit/src/main/groovy/org/shipkit/internal/notes/format/DetailedFormatter.java [hunk 2]
```diff
@@ -60,20 +62,21 @@ public String formatReleaseNotes(Collection<ReleaseNotesData> data) {
...
-                : buildHeader(version, date, "**", "**");
+    static String header(String version, boolean emphasizeVersion) {
+        return emphasizeVersion ? buildHeader(version, "# ", "")
+            : buildHeader(version, "#### ", "");
     }
 
-    private static String buildHeader(String version, Date date, String prefix, String postfix) {
-        return prefix + version + " (" + DateUtil.formatDate(date) + ")" + postfix + " - ";
+    private static String buildHeader(String version, String prefix, String postfix) {
+        return prefix + version + postfix;
```
Comment (wwilk) [inline subprojects/shipkit/src/main/groovy/org/shipkit/internal/notes/format/DetailedFormatter.java]: Seems that postfix is always "" so it's no longer necessary.

File: subprojects/shipkit/src/main/groovy/org/shipkit/internal/notes/format/DetailedFormatter.java [hunk 3]
```diff
@@ -60,20 +62,21 @@ public String formatReleaseNotes(Collection<ReleaseNotesData> data) {
...
     }
 
-    static String releaseSummary(String version, ContributionSet contributions, Map<String, Contributor> contributors,
-                                 String vcsCommitsLink, String publicationRepository) {
-        return authorsSummary(contributions, contributors, vcsCommitsLink) +
-                " - published to " + getBintrayBadge(version, publicationRepository) + "\n" +
-                authorsSummaryAppendix(contributions, contributors);
+    static String releaseSummary(Date date, String version, ContributionSet contributions, Map<String, Contributor>
+        contributors, String vcsCommitsLink, String publicationRepository) {
+        return " - " + "(" + DateUtil.formatDate(date) + ") "
```
Comment (wwilk) [inline subprojects/shipkit/src/main/groovy/org/shipkit/internal/notes/format/DetailedFormatter.java]: " - " + "(" can be just " - (". 
Also the format is different than the one suggested by the issue submitted. It is:
`(2017-11-20) n commits`

and should be:
`2017-11-20 - n commits`

Let's put this date prefix into private method.

**Comment by wwilk** [comment]
Thanks for the contribution!

**Comment by micd** [comment]
My pleasure!

**Pull request merged**